sk0-back/src/main.ts

```diff
@@ -1,8 +1,17 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
+import * as dotenv from 'dotenv';
+import * as fs from 'fs';
 
 async function bootstrap() {
+  const env = process.env.NODE_ENV || 'development';
+  const envConfig = dotenv.parse(fs.readFileSync(`.env.${env}`));
+  for (const k in envConfig) {
+    process.env[k] = envConfig[k];
+  }
   const app = await NestFactory.create(AppModule);
-  await app.listen(3000);
+  const port = process.env.PORT;
+  await app.listen(port);
```

Comment:
I have tried to start in a docker container with the env PORT set to 3005 in the Dockerfile, but didn't start for me

Reply:
I sort of fixed this, but when I tried for some reason it started listening using ipv6 addressing, and the page it linked to didn't respond, so I had to hard code to open localhost, which is absolutely horrible, I'll need some help tomorrow with this